fastplotlib/graphics/features/_mesh.py

```diff
@@ -116,3 +116,91 @@ def set_value(
 
         event = GraphicFeatureEvent(type=self._property_name, info={"value": value})
         self._call_event_handlers(event)
+
+
+def surface_data_to_mesh(data: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+    """
+    surface data to mesh positions and indices
+
+    expects data that is of shape: [m, n, 3] or [m, n]
+    """
+
+    data = np.asarray(data)
+
+    if data.ndim == 2:
+        # "image" of z values passed
+        # [m, n] -> [n_vertices, 3]
+        y = np.arange(data.shape[0]).reshape(data.shape[0], 1).repeat(data.shape[1], axis=1)
+        x = np.arange(data.shape[1]).reshape(1, data.shape[1]).repeat(data.shape[0], axis=0)
+        positions = np.column_stack((x.ravel(), y.ravel(), data.ravel()))
+    else:
+        if data.ndim != 3:
+            raise ValueError(
+                f"expect data that is of shape: [m, n, 3], [m, n]\n"
+                f"you passed: {data.shape}"
+            )
+        if data.shape[2] != 3:
+            raise ValueError(
+                f"expect data that is of shape: [m, n, 3], [m, n]\n"
+                f"you passed: {data.shape}"
+            )
+
+        # [m, n, 3] -> [n_vertices, 3]
+        positions = data.reshape(-1, 3)
+
+    # Create faces
+    w = data.shape[1]
+    i = np.arange(data.shape[0] - 1)
+    j = np.arange(w - 1)
+
+    j, i = np.meshgrid(j, i, indexing="ij")
+    start = j.ravel() + w * i.ravel()
+
+    indices = np.column_stack([start, start + 1, start + w + 1, start + w])
```

Comment:
I converted the for loop to be fully numpy, this is about 30% faster, does it look right to you? I think it is.